workbench-core/accounts/src/services/costCenterService.test.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+jest.mock('uuid', () => ({ v4: () => 'someId' }));
+
+import { DynamoDBClient, GetItemCommand, UpdateItemCommand } from '@aws-sdk/client-dynamodb';
+import { marshall } from '@aws-sdk/util-dynamodb';
+import { resourceTypeToKey } from '@aws/workbench-core-base';
+import { mockClient } from 'aws-sdk-client-mock';
+import Account from '../models/account';
+import CostCenter from '../models/costCenter';
+import CreateCostCenter from '../models/createCostCenter';
+import CostCenterService from './costCenterService';
+
+describe('CostCenterService', () => {
+  const ORIGINAL_ENV = process.env;
+  let accountMetadata: Account;
+  const costCenterService = new CostCenterService({ TABLE_NAME: 'tableName' });
+  const accountId = 'acc-someId';
+  const mockDynamo = mockClient(DynamoDBClient);
+
+  beforeEach(() => {
+    jest.resetModules();
+    process.env = { ...ORIGINAL_ENV };
+    process.env.AWS_REGION = 'us-east-1';
+    process.env.STACK_NAME = 'swb-swbv2-va';
+
+    accountMetadata = {
+      error: undefined,
+      id: accountId,
+      cidr: '',
+      hostingAccountHandlerRoleArn: '',
+      envMgmtRoleArn: 'sampleEnvMgmtRoleArn',
+      vpcId: 'vpc-123',
+      subnetId: 'subnet-123',
+      encryptionKeyArn: 'sampleEncryptionKeyArn',
+      environmentInstanceFiles: '',
+      stackName: `${process.env.STACK_NAME!}-hosting-account`,
+      status: 'CURRENT',
+      awsAccountId: 'awsAccountId',
+      externalId: 'externalId'
+    };
+  });
+
+  afterAll(() => {
+    process.env = ORIGINAL_ENV; // Restore old environment
+  });
+
+  describe('getCostCenter', () => {
+    describe('when a cost center has an id', () => {
+      let costCenterId: string;
+      let expectedCostCenter: CostCenter | undefined;
+
+      beforeEach(() => {
+        costCenterId = 'cc-someId';
+
+        expectedCostCenter = {
+          awsAccountId: accountMetadata.awsAccountId,
+          encryptionKeyArn: accountMetadata.encryptionKeyArn,
+          envMgmtRoleArn: accountMetadata.envMgmtRoleArn,
+          environmentInstanceFiles: accountMetadata.environmentInstanceFiles,
+          externalId: accountMetadata.externalId,
+          hostingAccountHandlerRoleArn: accountMetadata.hostingAccountHandlerRoleArn,
+          subnetId: accountMetadata.subnetId,
+          vpcId: accountMetadata.vpcId,
+          name: 'a name',
+          dependency: accountId,
+          description: 'a description',
+          id: costCenterId
+        };
+      });
+
+      describe('and the cost center has been saved', () => {
```

Comment:
It breaks down the requirements for the test, improving modularity and removing lots of code duplication between tests. Another great part is that it allows you to manage different configurations in different `beforeEach` methods. You're effectively stating each criteria you need for a test so that you don't have a super long test description.